sourcegraph/jetbrains
# Add cody settings editor

Fixes CODY-1310

## Changes

This PR adds cody settings editor with schema validation and code completion.

![image](https://github.com/user-attachments/assets/c8d0cd6c-4cb4-44a8-93b2-2aca35855a0d)
![image](https://github.com/user-attachments/assets/bfa03906-fffb-4ffd-8adf-781885ebca2e)
![image](https://github.com/user-attachments/assets/5db17148-f281-461a-94c9-1cf42a7e6e77)

## Test plan

1. Build this PR with CODY_DIR set to [this PR branch](https://github.com/sourcegraph/cody/pull/5080).
2. Run action `Open Cody Settings Editor` (you can use `shift shift` to find it)
3. Change some Cody settings. E.g. you can try to disable autocomplete:
`"cody.autocomplete.enabled": true`
4. Hit `Ctrl + S` to trigger the config update
5. Try to use autocomplete in some other file and make sure it does not work.
6. Revert your changes and make sure autocomplete is enabled again.
